wrapper/pycc/src/qcc_db/ccObject.cpp

```diff
@@ -134,6 +142,15 @@ void define_ccObject(py::module &m)
         .def("swapChildren", &ccHObject::swapChildren, "firstChildIndex"_a, "secondChildIndex"_a)
         .def("getChildIndex", &ccHObject::getChildIndex, "child"_a)
         .def("getIndex", &ccHObject::getIndex)
+        .def(
+            "addChild",
+            [](py::object &self, py::object &child, ccHObject::DEPENDENCY_FLAGS dependencyFlags, int insertIndex) {
```

Comment:
Yes, but the cast-like syntax is use everywhere else in the code. Should we change these occurences too ?

Reply:
I think it will be hard to comply with CC cparent-children dependency management.
for example
```
parent = pycc.ccPointCloud("parent")
child = pycc.ccPointCloud("child")

parent.addChild(child, pycc.DEPENDENCY_FLAGS.DP_PARENT_OF_OTHER) #it's default but it's implicit
del parent
print(child.getName())# segfault, shoud throw NameError
```

It segfault even with keep_alive<1,2>

but fortunately something like this is ok
```
parent = pycc.ccPointCloud("parent")
child = pycc.ccPointCloud("child")

parent.addChild(child, pycc.DEPENDENCY_FLAGS.DP_PARENT_OF_OTHER) #it's default but it's implicit
print(parent.getChildNumber())# shows 1
del child
print(parent.getChildNumber())# shows 0
```

Reply:
> It segfault even with keep_alive<1,2>

That is because I think you reversed the keep_alive rule, it should be 

```
py::keep_alive<2, 1>() // keep alive parent (1) while added child (2) is alive
```

With that none of the 2 example segfault

Reply:
You are right. both test case are ok in some extends. In one hand, It's clearly better because it should not crash nor lead to UB. But in another hand, I think it does not match completely the dependency policy of CC because if policy is DP_DELETE_OTHER or DP_PARENT_OF_OTHER, when you call `del parent` `child` should be deleted as well (so calling `child` or any method of `child` should throw a NameError). With keep_alive<2,1>` child` is not deleted and still accessible. Maybe it has some logic because child was created via the Python Runtime so we can consider it has ownership. What is your opinion on that?

Reply:
The thing is that to have a `NameError` we would need to delete the variable from from the interpreter, which is probably
tricky because we don't have access to the scope that contains the variable to be deleted.

So having something that does not crash (and should not lead to memory leak) is better than having crash i think,  especially since this plugin is meant to be used by people who are not 100% software engineers